x-pack/protocol/src/main/java/org/elasticsearch/protocol/xpack/license/PostStartTrialResponse.java

```diff
@@ -0,0 +1,236 @@
+import static org.elasticsearch.common.xcontent.ConstructingObjectParser.optionalConstructorArg;
+
+public class PostStartTrialResponse extends AcknowledgedResponse {
+
+    private static final ParseField TRIAL_WAS_STARTED_FIELD = new ParseField("trial_was_started");
+    private static final ParseField TYPE_FIELD = new ParseField("type");
+    private static final ParseField ERROR_MESSAGE_FIELD = new ParseField("error_message");
+    private static final ParseField ACKNOWLEDGE_DETAILS_FIELD = new ParseField("acknowledge");
+    private static final ParseField ACKNOWLEDGE_HEADER_FIELD = new ParseField("message");
+
+    private static final ConstructingObjectParser<PostStartTrialResponse, Void> PARSER = new ConstructingObjectParser<>(
+        "post_start_trial_response",
+        true,
+        (arguments, aVoid) -> {
+            boolean acknowledged = (boolean) arguments[0];
+            boolean trialWasStarted = (boolean) arguments[1];
+            String type = (String) arguments[2];
+            String errorMessage = (String) arguments[3];
+            @SuppressWarnings("unchecked")
+            Tuple<String, Map<String, String[]>> acknowledgeDetails = (Tuple<String, Map<String, String[]>>) arguments[4];
+
+            if (trialWasStarted) {
+                return new PostStartTrialResponse(acknowledged, type);
+            } else {
+                return new PostStartTrialResponse(acknowledged, errorMessage, acknowledgeDetails.v1(), acknowledgeDetails.v2());
+            }
+        }
+    );
+
+    static {
+        declareAcknowledgedField(PARSER);
+        PARSER.declareBoolean(constructorArg(), TRIAL_WAS_STARTED_FIELD);
+        PARSER.declareString(optionalConstructorArg(), TYPE_FIELD);
+        PARSER.declareString(optionalConstructorArg(), ERROR_MESSAGE_FIELD);
+        PARSER.declareObject(optionalConstructorArg(), (parser, v) -> {
+            Map<String, String[]> acknowledgeMessages = new HashMap<>();
+            String message = null;
+            XContentParser.Token token;
+            String currentFieldName = null;
+            while ((token = parser.nextToken()) != XContentParser.Token.END_OBJECT) {
+                if (token == XContentParser.Token.FIELD_NAME) {
+                    currentFieldName = parser.currentName();
+                } else {
+                    if (currentFieldName == null) {
+                        throw new XContentParseException(parser.getTokenLocation(), "expected message header or acknowledgement");
+                    }
+                    if ("message".equals(currentFieldName)) {
+                        if (token != XContentParser.Token.VALUE_STRING) {
+                            throw new XContentParseException(parser.getTokenLocation(), "unexpected message header type");
+                        }
+                        message = parser.text();
+                    } else {
+                        if (token != XContentParser.Token.START_ARRAY) {
```

Comment:
Sure thing, I shamelessly stole this from https://github.com/elastic/elasticsearch/pull/32214/files#diff-a651ec0801020d94344b5d3c51ec39d0R63 so it might be worth sharing between the two